cmd/cosign/cli/fulcio/fulcio.go

```diff
@@ -62,27 +65,54 @@ func (rf *realConnector) OIDConnect(url, clientID, secret, redirectURL string) (
 	return oauthflow.OIDConnect(url, clientID, secret, redirectURL, rf.flow)
 }
 
-func getCertForOauthID(priv *ecdsa.PrivateKey, fc api.LegacyClient, connector oidcConnector, oidcIssuer, oidcClientID, oidcClientSecret, oidcRedirectURL string) (*api.CertificateResponse, error) {
-	pubBytes, err := x509.MarshalPKIXPublicKey(&priv.PublicKey)
-	if err != nil {
-		return nil, err
-	}
-
+func getCertForOauthID(sv signature.SignerVerifier, fc api.LegacyClient, connector oidcConnector, oidcIssuer, oidcClientID, oidcClientSecret, oidcRedirectURL string) (*api.CertificateResponse, error) {
 	tok, err := connector.OIDConnect(oidcIssuer, oidcClientID, oidcClientSecret, oidcRedirectURL)
 	if err != nil {
 		return nil, err
 	}
 
-	// Sign the email address as part of the request
 	h := sha256.Sum256([]byte(tok.Subject))
-	proof, err := ecdsa.SignASN1(rand.Reader, priv, h[:])
-	if err != nil {
-		return nil, err
+
+	var pubBytes, proof []byte
+	var algorithm string
+	switch sv := sv.(type) {
+	case *signature.RSAPKCS1v15SignerVerifier:
```

Comment:
To simplify this, can you call https://github.com/sigstore/sigstore/blob/main/pkg/cryptoutils/publickey.go#L77 on the the signer's `PublicKey`? (Fulcio doesn't currently support keys provided as raw PKCS1, so they should all be either PKIX or PEM-encoded PKIX or PKCS1 - https://github.com/sigstore/fulcio/blob/main/pkg/challenges/challenges.go#L84 - I'm filing a bug to make Fulcio more permissive in what it supports)

Another thing that's undocumented is Fulcio actually ignores the algorithm field, so you should be able to simplify this to not use a case statement and just treat all key algorithms as the same. You can then just not set `Algorithm` in `api.Key`